[HttpClient] Add `error.type` for traces and metrics (#5005)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md

  `http` or `http/dup`.
  ([#5003](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5003))

* An additional attribute `error.type` will be added to activity and
  `http.client.request.duration` metric in case of failed requests as per the
  [specification](https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md#common-attributes).

  Users moving to `net8.0` or newer frameworks from lower versions will see

>   difference in values in case of an exception. `net8.0` or newer frameworks adds

Reviewer: ```suggestion
  difference in values in case of an exception. `net8.0` or newer frameworks add
```

>   the ability to further drilldown the exceptions to a specific type through

Reviewer: ```suggestion
  the ability to further drill down the exceptions to a specific type through
```

  [HttpRequestError](https://learn.microsoft.com/dotnet/api/system.net.http.httprequesterror?view=net-8.0)

>   enum. For lower versions the individual types will be rolled in to a single

Reviewer: ```suggestion
  enum. For lower versions, the individual types will be rolled in to a single
```
